### ISSUE TITLE
New Color Format: YUY2

### ISSUE THREAD

**Issue by martin31821** [libjpeg-turbo/libjpeg-turbo#390]
I'm reading out YUY2 data from a capture card with a memory layout like this: 
`y0 cb0 y1 cr0 y2 cb1 y3 cr1` where a 4:2:2 subsampling is used.

I'd like to see this data format as input and output format for libjpeg-turbo, as it is frequently used and shouldn't be particular hard to parse, given that YCbCr code is already contained within libjpeg.
A simple, non optimized conversion code might be found here: https://gist.github.com/royshil/fa98604b01787172b270.

**Comment by dcommander** [comment]
Referring to https://github.com/libjpeg-turbo/libjpeg-turbo/blob/master/.github/CONTRIBUTING.md, libjpeg-turbo implements only algorithms that are necessary for JPEG compression and decompression.  Any other algorithm is out-of-scope by definition, since libjpeg-turbo is not and never has been a comprehensive image processing or conversion toolkit.

libjpeg-turbo (both APIs-- the libjpeg and TurboJPEG API) supports YUV encoding and decoding only because it can do so by exposing the underlying JPEG color conversion algorithms.  This is not an independent image processing/conversion feature.  It is simply exposing the underlying planar YCbCr<-->packed RGB color conversion routines used in JPEG compression/decompression, or it is shortcutting the JPEG pipeline to bypass those color conversion routines.  Packed YCbCr/YUV is not part of JPEG compression/decompression, so it would be necessary to implement a whole new interface for it, design SIMD conversion algorithms, etc.  Because of how the libjpeg API is structured, it would not be possible to simply feed a YUY2 image directly into the SIMD color converter, even if a YUY2-capable SIMD color converter existed.  It would be necessary to convert YUY2 into planar YCbCr first, and there is no reason why our library would do a better job at that than an external library.

**Comment by dcommander** [comment]
Note that the libjpeg API does accept packed YCbCr in un-subsampled form, i.e. Y0 Cb0 Cr0 Y1 Cb1 Cr1, etc., but this has no direct equivalent in video colorspaces, nor is the packed-to-planar conversion algorithm SIMD-accelerated on most platforms.
